### DISCUSSION TITLE
Anyone getting encoders to work, also have an LED question

### DISCUSSION THREAD

**Discussion by Babylonrising** [shanteacontrols/OpenDeck#283] [Q&A]
Hey guys! New member here, looking to buy an L board but I have a question before I sink a bunch of money, has anyone got encoders to work smoothly and without having to turn them 5 times, or without accelerate making them jump up 5 notes in a small nudge. Also with if I'm going to setup a pushbutton to select a solo am I better off just tying in a led to my push button breakout board with its own resistor going into the opendeck or having the led go directly into the open deck? I want it to light up when I have solo pressed on my bluebox mixer. Thank you in advance everyone. Also just some backroom on why I need encoders. I'm going to be using them to control envelopes within my eurorack nerdseq sequencer and fh-2 module so I jump around between 4 different voices so I can't use potentiometers

**Comment by paradajz** [comment]
> Hey guys! New member here, looking to buy an L board but I have a question before I sink a bunch of money, has anyone got encoders to work smoothly and without having to turn them 5 times

I have no idea what do you mean with the last part. If you get good quality encoders you won't have any problems. The one I always use is mentioned here:
https://github.com/shanteacontrols/OpenDeck/wiki/Recommended-components-and-where-to-get-them#encoders

> (...) tying in a led to my push button breakout board with its own resistor going into the opendeck or having the led go directly into the open deck?

The wiki cleary mentiones that no resistor is required on L board:
https://github.com/shanteacontrols/OpenDeck/wiki/Official-OpenDeck-L-boards#single-color-leds

**Reply by Babylonrising** [reply]
Oh I thought c was the ground, ok so then column! So then if what I have marked as ground is going to c on the board is my pcb designed correctly? Also with the leds can I program them within the  configuration app to light up when I have a button mapped to solo presses?

I believe those encoders you mentioned are the ones used to make that pcb, I heard from other people on here that you have to turn them many times to go from 0-127, last I saw you said you might write a more advanced accelerate code to make encoders to work better, also can you recommend a good push encoder as I'll be using four of those

**Reply by paradajz** [reply]
So far every single question you have asked is answered already in the wiki. Please read it.
I don't have recommendation for encoder with push button right now.

**Reply by Babylonrising** [reply]
Because before I buy your product for almost $200 then invest in another like $300 in parts, and design a pcb I just want two questions confirmed before I purchase as I stated I'm totally new to this and some of this confusing, that's the point of this section, for people to ask questions they still don't understand. I want to make a custom controller while still supporting small business, so please just help if you can vs telling me to go read something I already did but still find a bit confusing

**Reply by paradajz** [reply]
Before you buy something for 200$ you should read as much as possible about it, correct? Besides, a simple answer here is to buy Raspberry Pi Pico for few bucks and load pre-build OpenDeck firmware on it to experiment and then see if the L board is for you or not.

https://github.com/shanteacontrols/OpenDeck/wiki/Flashing-RP2040-boards

**Reply by Babylonrising** [reply]
Ohhh that's a good idea thanks! I did read alotttttt but those two things I asked I still find a bit confusing if you could help out with an answer

**Reply by Babylonrising** [reply]
Actually! I just printed out a picture of the L board and used a marker to connect pins on paper and understand it now! Still can't figure out my led question
